tests/test_matrix_utils_sync_bootstrap_login.py

```
@@ -673,13 +679,66 @@ async def test_login_matrix_bot_e2ee_store_path_creation(
+        patch("mmrelay.matrix_utils.is_e2ee_enabled", return_value=True),
+        patch(
+            "mmrelay.matrix_utils._ensure_own_device_cross_signed",
+            ensure_cross_signed,
+        ),
+        patch(
+            "mmrelay.matrix_utils.get_e2ee_store_dir",
+            return_value=TEST_E2EE_STORE_PATH,
+        ),
+        patch("os.makedirs"),
+    ):
+        with pytest.raises(asyncio.CancelledError):
```

Ruff (0.16.0)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)